docs/DSA-Problem-Solution/Merge_K_Sorted_Arrays.md

```diff
@@ -0,0 +1,100 @@
+---
+id: merge-k-sorted-arrays
+title: Merge K Sorted Arrays
+sidebar_label: GFG
+tags: [GFG, Arrays, Sorting, Heap, DSA]
+description: Given k sorted arrays with each of size k arranged in the form of a matrix of size k * k. The task is to merge them into one sorted array.
+---
+
+# Partition Equal Subset Sum Algorithm (GFG)
+
+## Description
+
+The **Merge K Sorted Arrays** problem is an intuitive problem based on Priority Queue where we have to merge k sorted arrays into one final sorted array.
+
+### Problem Definition
+
+Given:
+
+- A 2D array of integers `arr` of size k\*k.
+
+Objective:
+
+- Merge these arrays into 1 sorted array. Return the merged sorted array.
+
+### Algorithm Overview
+
+1. **Min Heap Approach**:
+
+- Creating a `MinHeap` and Insert the `first` element `of all the k arrays`.
+- Remove the `top most` element of Minheap and `put` it in the output array.
+- And insert the `next` element `from`the `array of removed` elements.
+- To get the `result` the step must continue until there is no element left in the MinHeap.
+
+2. **Return** `result`, which is the final sorted array after merging k sorted arrays.
+
+### Time Complexity
+
+- **Time Complexity**: O(K^2\* log(K)), where insertion and deletion in a Min Heap requires log K time and for all K^2 elements it takes (K^2 \* log(K)) time
```

Comment:
replace `O(K^2\* log(K))` with `$O(K^2\* log(K))$`, `k^2` with `$k^2$`, and `(K^2 \* log(K))` with `$(K^2 \* log(K))$`